### PR TITLE
Machines can now be pried open multiple times and maintain their initial densities

### LINKED ISSUE #26833: Default_pry_open issues
Several machines use default_pry_open, which calls open_machine(), which while drops the contents of the machine, also makes them non-dense. This makes sense for sleepers, cryochambers and cyborg rechargers, but has problems in various other devices. 

- Gibber
- Holopad: holopads are already nondense so it isn't much of an issue but still
- Hydroponic trays
- Monkeyrecycler
- Ore Redemption
- Recycler
- Processor
- Seed Extractor
- Smartfridges
- Suit Storage Unit: if opened via the interface instead of a crowbar, it won't become nondense, but crowbar prying makes it so.

### PR DESCRIPTION
**Pull request by atosti**
## About The Pull Request
These changes fix how machines are pried open with crowbars. Currently, most machines can be pried open, but many of them have no method for being closed again. This means they can be pried once, and then never again (as their internal logic has them stuck in an "open" state).

Additionally, the densities of these machines is also inconsistent, as density is tied to the procs for opening/closing machines (open = non-dense, closed = dense). Thus, these new changes allow desired densities to be passed to `open_machine()` and `close_machine()`, as well as `default_pry_open()`, meaning that atypical machine densities can be maintained (e.g. machines that should remain dense when open, or non-dense when closed).

I've also added a `close_after_pry` boolean parameter to the `default_pry_open()` proc, which determines whether to immediately close a machine after opening it. This is useful for machines that don't really have a use case for remaining open, often lacking a sprite to represent this state as well. 
* Note: Opening and immediately closing machines with this boolean will still drop their contents onto the floor, but will now immediately "close" in their logic, allowing for further prying attempts in the future.

It's worth noting that this implements default density values for these procs, which match the existing behavior for machines, so as to (hopefully) not disrupt existing or expected machine behavior.

Two caveats to these changes currently exist:
1. On machines that immediately close after prying, the prying action can now be spammed to the chat with repeated clicking. I'm uncertain if this needs some sort of spam protection or if it's fine as is.
2. I've only been able to manually test this code. I'd love to write unit tests for it, as it affects a lot of different machines, but don't know where to begin with DM Unit Testing (or which files would be good examples to reference in the code base). 
    * Note: I did manually test each and every machine that calls `default_pry_open()` and they all seem to be working correctly. (Except for `obj/machinery/plumbing/sender`, but that doesn't seem to need prying, as it has no contents to drop, only reagents.)

As always, let me know if any improvements/changes should be made.

This closes #26833.

## Why It's Good For The Game
These changes allow crowbar prying to correctly occur multiple times on any machine, which is intended behavior. It prevents player confusion that could occur when a machine couldn't be pried open a second time during a shift, even though it had previously been pried before, forcing players to question themselves. (Are they missing something? Did they perform the action a different way last time? Is the machine actually still powered on instead of off? Etc.)

These changes also maintain the correct density for machines after prying, preventing scenarios where a machine might behave differently once it had been pried open. (An example of this was being able to walk through a smartfridge after prying it open.) Additionally, players are no longer required to know/use workarounds (such as machine disassembly) to retrieve a powered-off machine's contents.

Overall, these changes improve consistency around machines, creating more scenarios where they behave as players would expect.

## Changelog
:cl:
fix: machines can now be pried open more than once.
fix: machines now have the correct density when pried open.
/:cl:

### CODE REVIEW

**Comment by san7890** [comment]
image

this is what's failing CI

File: code/game/machinery/_machinery.dm [hunk 1]
```diff
@@ -326,10 +326,11 @@
  * Will update the machine icon and any user interfaces currently open.
  * Arguments:
  * * drop - Boolean. Whether to drop any stored items in the machine. Does not include components.
+ * * density - Boolean. Whether to make the object dense when it's open.
  */
-/obj/machinery/proc/open_machine(drop = TRUE)
+/obj/machinery/proc/open_machine(drop = TRUE, density = FALSE)
```
Comment (san7890) [inline code/game/machinery/_machinery.dm:331]: `density` is a byond-level variable, so this is unfortunately a bit confusing. can you change it to

```suggestion
/obj/machinery/proc/open_machine(drop = TRUE, density_to_set = FALSE)
```

I got confused thinking you were somehow changing the value of the byond-level variable in the arguments of a proc. that might be unreasonable but i got really flummoxed.

File: code/game/machinery/washing_machine.dm [hunk 2]
```diff
@@ -416,7 +416,7 @@ GLOBAL_LIST_INIT(dye_registry, list(
 		new /obj/item/stack/sheet/iron(drop_location(), 2)
 	qdel(src)
 
-/obj/machinery/washing_machine/open_machine(drop = 1)
+/obj/machinery/washing_machine/open_machine(drop = 1, density = FALSE)
```
Comment (san7890) [inline code/game/machinery/washing_machine.dm:419]:
```suggestion
/obj/machinery/washing_machine/open_machine(drop = TRUE, density = FALSE)
```

may as well fix it while you're here

File: code/game/machinery/washing_machine.dm [hunk 3]
```diff
@@ -416,7 +416,7 @@ GLOBAL_LIST_INIT(dye_registry, list(
...
 	..()
```
Comment (san7890) [inline code/game/machinery/washing_machine.dm:420]:
```suggestion
	. = ..()
```

ditto "while you're here" cleanup stuff

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 4]
```diff
@@ -365,7 +365,7 @@
 	flick("pod-open-anim", src)
 	..()
```
Comment (san7890) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm:366]:
```suggestion
	return ..()
```

ditto while your'e here

**Review by san7890 — CHANGES_REQUESTED** [review]
few nits, and some other things that would be well appreciated if you could pass in since it's very unlikely that someone will touch this code for a few years

**Comment by san7890** [comment]
> I'd love to write unit tests for it, as it affects a lot of different machines, but don't know where to begin with DM Unit Testing (or which files would be good examples to reference in the code base).

Try something like the surgery unit tests for a template, how it allocates humans and gives them the tools, then manually calls the procs using the allocated equipment by passing them in as the references... or maybe `/datum/unit_test/machine_disassembly`? unsure

File: code/game/machinery/_machinery.dm (hunk 1 above)
Comment (atosti) [reply code/game/machinery/_machinery.dm:331]: Gotcha, that makes sense. Should I also adjust all the child overrides to pass in a parameter with this new name? Or is that unnecessary?

**Comment by atosti** [comment]
Thanks for the guidance on where to find good Unit Test examples. I'll probably include them as part of another PR if I design them in the future. Let me know if further changes are needed.

**Pull request merged**